gk/main.c

```diff
@@ -542,6 +604,42 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	return ret;
 }
 
+static int
+drop_flow_entry_randomly(
+	struct gk_instance *instance, unsigned int flow_ht_size)
+{
+	int  ret;
+	uint32_t entry_id = rte_rand() % flow_ht_size;
+	uint32_t entry_start_idx = entry_id;
+	struct flow_entry *fe;
+
+	while (true) {
+		fe = &instance->ip_flow_entry_table[entry_id];
+		if (fe->last_active_ts == 0) {
+			entry_id = (entry_id + 1) % flow_ht_size;
+			fe = NULL;
+		} else
+			break;
+
+		if (entry_start_idx == entry_id)
+			break;
```

Comment:
I think the code can be simplified by just returning -1 here instead of breaking. That way, we don’t need to keep making sure fe is set to NULL in the case above, and don’t need the separate if statement after the loop.